SPEC.md

```diff
@@ -2997,13 +2997,17 @@ A WDL task can be thought of as a template for running a set of commands - speci
 
 A task is defined using the `task` keyword, followed by a task name that is unique within its WDL document.
 
-A task has a required [`command`](#command-section) that is a template for a Bash script.
+Tasks are comprised of the following elements:
 
-Tasks explicitly define their [`input`s](#task-inputs) and [`output`s](#task-outputs), which is essential for building dependencies between tasks and workflows. The value of an input declaration may be supplied by the caller. Tasks may have additional "private" declarations within the task body. All task declarations may be initialized with hard-coded literal values, or may have their values constructed from expressions. Input and private declarations can be referenced in the command template.
+* A singular, optional [`input`](#task-inputs) section, which defines the inputs for the task.
+* A singular, required [`command`](#command-section), which defines the Bash script to be executed.
+* A singular, optional [`output`](#task-outputs) section, which defines the outputs for the task.
+* A singular, optional [`runtime`](#runtime-section) section, which defines the required runtime environment conditions.
+* A singular, optional [`meta`](#metadata-sections) section, which defines task-level metadata.
+* A singular, optional [`parameter_meta`](#parameter-metadata-section) section, which defines parameter-level metadata.
+* Any number of [private declarations](#private-declarations).
 
-A task may also specify [requirements for the runtime environment](#runtime-section) (such as the amount of RAM or number of CPU cores) that must be satisfied in order for its commands to execute properly.
-
-There are two optional metadata sections: the [`meta`](#metadata-sections) section, for task-level metadata, and the [`parameter_meta`](#parameter-metadata-section) section, for parameter-level metadata.
+Although there is no enforced order for task elements, it is recommended to organize them logically, starting with execution-based elements in chronological order, then specifying runtime requirements, and finally, adding metadata about the task.
```

Comment:
This seems more suited to a style guide. I don't think it belongs in the spec.

I might support more neutral language such as "...it is recommended to follow a consistent pattern for all the tasks in a workflow."

Reply:
I now realize this conflicts with @geoffjentry comment above. I'll bring this up on Slack and try to get consensus.

Reply:
@jdidion Agree a style guide would be more appropriate.

Reply:
I have addressed these by simply stating there is no order to the elements.